sensor/debugger/k8s/fake.go

```diff
@@ -318,9 +400,56 @@ func (f *FakeEventsManager) createEvent(msg resources.InformerK8sMsg, ch chan<-
 }
 
 // waitOnMode waits depending on the mode
-func (f *FakeEventsManager) waitOnMode() {
+func (f *FakeEventsManager) waitOnMode(events []string) error {
 	switch f.Mode {
 	case Delay:
 		time.Sleep(f.Delay)
+		return nil
+	case ChannelAck:
+		if len(events) == 0 {
+			return nil
+		}
+		receivedEvents := 0
+		for {
+			timeout := time.After(5 * time.Second)
+			select {
+			case <-timeout:
+				return errors.New("timeout reached waiting for event")
+			case event := <-f.AckChannel:
+				eventFound, err := isEventInSlice(event, events)
```

Comment:
`isEventInSlice` will return false if the `event` is empty. I think we don't need to change anything there.

Reply:
Yes, it will return false but it also returns an error and this may be not what we want. Maybe you could check whether the object parsed from json is not 0-valued?

Reply:
I'm not sure if I'm understanding what you're saying 🤔 We are not parsing the event received in the channel. We parse the list of events read from the trace file. The event received is passed to the `compareFunc` where we read the specific resource (e.g. `event.GetNamespace()`) This function will return nil if the event is empty or nil. So, at the end we return `false, nil` if the event is empty. We're only returning an error here if the trace file contains events that cannot be parsed.

Reply:
Oh, many thanks for the explanation, now this is clear :)

Now I wonder if there is a way to not parse the entire array on each message but to do it once somewhere earlier - do you think this would be possible? (separate PR is okay)

Reply:
Thanks, it looks good :) It could be even a separate function ;)

Reply:
Done 🙂

Reply:
Thanks!